astropy/visualization/lupton_rgb.py

```diff
@@ -0,0 +1,469 @@
+            raise ValueError("please specify either a single image "
+                             "or red, green, and blue images.")
+        return image_r
+
+    intensity = (image_r + image_g + image_b)/3.0
+
+    # Repack into whatever type was passed to us
+    return np.array(intensity, dtype=image_r.dtype)
+
+
+class Mapping(object):
+    """Baseclass to map red, blue, green intensities into uint8 values."""
+
+    def __init__(self, minimum=None, image=None):
+        """
+        Create a mapping.
+
+        Parameters
+        ----------
+        minimum : float or sequence(3)
+            Intensity that should be mapped to black (a scalar or array for R, G, B).
+        image : `~numpy.ndarray`, optional
+            The image to be used to calculate the mapping.
+            If provided, it is also used as the default for make_rgb_image().
+        """
+        self._uint8Max = float(np.iinfo(np.uint8).max)
+
+        try:
+            len(minimum)
+        except TypeError:
+            minimum = 3*[minimum]
+        if len(minimum) != 3:
+            raise ValueError("please provide 1 or 3 values for minimum.")
+
+        self.minimum = minimum
+        self._image = np.asarray(image)
+
+    def make_rgb_image(self, image_r=None, image_g=None, image_b=None,
+                       x_size=None, y_size=None, rescale=None):
+        """
+        Convert 3 arrays, image_r, image_g, and image_b into an 8-bit RGB image.
+
+        Parameters
+        ----------
+        image_r : `~numpy.ndarray`, optional
+            Image to map to red (if None, use the image passed to the
+            constructor).
+        image_g : `~numpy.ndarray`, optional
+            Image to map to green (if None, use image_r).
+        image_b : `~numpy.ndarray`, optional
+            Image to map to blue (if None, use image_r).
+        x_size : int, optional
+            Desired width of RGB image (or None).  If y_size is None, preserve
+            aspect ratio.
+        y_size : int, optional
+            Desired height of RGB image (or None).
+        rescale : float, optional
+            Make size of output image rescale*size of the input image.
+            Cannot be specified if x_size or y_size are given.
+
+        Returns
+        -------
+        RGBimage : `~numpy.ndarray`
+            RGB (integer, 8-bits per channel) color image as an NxNx3 numpy array.
+        """
+        if image_r is None:
+            if self._image is None:
+                raise RuntimeError("you must provide an image or pass one "
+                                   "to the constructor.")
+            image_r = self._image
+        else:
+            image_r = np.asarray(image_r)
+
+        if image_g is None:
+            image_g = image_r
+        else:
+            image_g = np.asarray(image_g)
+
+        if image_b is None:
+            image_b = image_r
+        else:
+            image_b = np.asarray(image_b)
```

Comment:
I played with an earlier version of your code for JWST Data Analysis Forum and actually have some examples at http://astroimtools.readthedocs.io/en/latest/astroimtools/notebooks/make_rgb.html (which I intend to update to use this PR once it is merged). In that example, we played with using only 2 images as inputs, while setting G to None. The result was not satisfactory, so we ended up using the average of R and B as G. What I am trying to say is that, I agree in requiring exactly 3 images, because that was what we ended up using anyway even when we only had 2 colors. Does that make sense?